plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java

```diff
@@ -91,4 +100,39 @@ public static String replaceVariables(@NotNull String name) {
                 return System.getProperty(name);
         }
     }
+    
+    /**
+     * Appends <code>--host</code> and <code>--port</code> parameters from connection settings to the list of command parameters
+     */
+    public static void addHostAndPortParamsToCmd(@NotNull DBPDataSourceContainer dataSourceContainer, @NotNull List<String> cmd) {
+        DBPConnectionConfiguration connectionInfo = dataSourceContainer.getActualConnectionConfiguration();
+        String hostname = "";
+        String port = "";
+        if (connectionInfo.getConfigurationType().equals(DBPDriverConfigurationType.MANUAL)) {
+            hostname = connectionInfo.getHostName();
+            port = connectionInfo.getHostPort();
+        } else {
+            String url = connectionInfo.getUrl();
+            String sampleUrl = dataSourceContainer.getDriver().getSampleURL();
+            if (sampleUrl != null) {
+                DBPConnectionConfiguration config = JDBCURL.extractConfigurationFromUrl(sampleUrl, url);
+                if (config != null) {
+                    hostname = config.getHostName();
+                    port = config.getHostPort();
+                }
+            } else {
+                try {
+                    URI uri = URI.create(url);
+                    hostname = uri.getHost();
+                    port = String.valueOf(uri.getPort());
+                } catch (Exception e) {
+                    log.error("Can't parse connection URL", e);
+                }
+            }
+        }
+        cmd.add("--host=" + hostname);
```

Comment:
What if `hostname` is empty, too?

Reply:
It was not supposed to be empty. I decided to let this behaviour be untouched.

Reply:
Actually, I think it's not possible as it's not a valid URI anymore.